### PR TITLE
Update Sortable.php

### PR DESCRIPTION
**Pull request by sammyschnei**
Eliminate hard-coded 'id' as primary database key. Check for $primaryKey defined in the model, default to 'id'.

### CODE REVIEW

**Comment by LukeTowers** [comment]
This is in the wrong branch, any pull requests need to be submitted to the develop branch instead.

**Comment by sammyschnei** [comment]
Sorry, new to this. Changed to Octobercms:develop.

File: src/Database/Traits/Sortable.php [hunk 1]
```diff
@@ -57,9 +57,10 @@ public function setSortableOrder($itemIds, $itemOrders = null)
             throw new Exception('Invalid setSortableOrder call - count of itemIds do not match count of itemOrders');
         }
 
+        $primaryKey = (!empty($this->primaryKey) ? $this->primaryKey : 'id');
         foreach ($itemIds as $index => $id) {
             $order = $itemOrders[$index];
-            $this->newQuery()->where('id', $id)->update([$this->getSortOrderColumn() => $order]);
+            $this->newQuery()->where($primaryKey, $id)->update([$this->getSortOrderColumn() => $order]);
```
Comment (daftspunk) [inline src/Database/Traits/Sortable.php]: This line should read `$this->newQuery()->where($this->getKeyName(), $id)->update([$this->getSortOrderColumn() => $order]);`

File: src/Database/Traits/Sortable.php [hunk 2]
```diff
@@ -57,9 +57,10 @@ public function setSortableOrder($itemIds, $itemOrders = null)
             throw new Exception('Invalid setSortableOrder call - count of itemIds do not match count of itemOrders');
         }
 
+        $primaryKey = (!empty($this->primaryKey) ? $this->primaryKey : 'id');
```
Comment (daftspunk) [inline src/Database/Traits/Sortable.php]: Remove this line

File: src/Database/Traits/Sortable.php (hunk 1 above)
Comment (sammyschnei) [inline src/Database/Traits/Sortable.php]: Ok, I think I put this update in the right place... Should be all set now.

**Pull request merged**